diff --git a/sites/all/modules/contrib/admin_menu/admin_menu.inc b/sites/all/modules/contrib/admin_menu/admin_menu.inc
index 27dc7d1..36eb2da 100644
--- a/sites/all/modules/contrib/admin_menu/admin_menu.inc
+++ b/sites/all/modules/contrib/admin_menu/admin_menu.inc
@@ -364,7 +364,7 @@ function admin_menu_expand_args($arguments) {
  */
 function admin_menu_links_menu($tree) {
   $links = array();
-  foreach ($tree as $data) {
+  foreach ($tree as $key => $data) {
     // Skip invisible items.
     if (!$data['link']['access'] || $data['link']['type'] == MENU_CALLBACK) {
       continue;
@@ -386,14 +386,14 @@ function admin_menu_links_menu($tree) {
     // Remove description to prevent mouseover tooltip clashes.
     unset($data['link']['localized_options']['attributes']['title']);
 
-    $links[$data['link']['mlid']] = array(
+    $links[$key] = array(
       '#title' => $data['link']['title'],
       '#href' => $data['link']['href'],
       '#options' => $data['link']['localized_options'],
       '#weight' => $data['link']['weight'],
     );
     if ($data['below']) {
-      $links[$data['link']['mlid']] += admin_menu_links_menu($data['below']);
+      $links[$key] += admin_menu_links_menu($data['below']);
     }
   }
   return $links;
diff --git a/sites/all/modules/contrib/admin_menu/admin_menu.module b/sites/all/modules/contrib/admin_menu/admin_menu.module
index 123faa4..d4cadef 100644
--- a/sites/all/modules/contrib/admin_menu/admin_menu.module
+++ b/sites/all/modules/contrib/admin_menu/admin_menu.module
@@ -410,8 +410,6 @@ function admin_menu_output() {
     $content['menu']['#theme'] = 'admin_menu_links';
     // Ensure the menu tree is rendered between the icon and user links.
     $content['menu']['#weight'] = 0;
-    // Do not sort the menu tree, since it is sorted already.
-    $content['menu']['#sorted'] = TRUE;
 
     // Add menu additions.
     $content['icon'] = admin_menu_links_icon();
@@ -436,6 +434,48 @@ function admin_menu_output() {
 }
 
 /**
+ * Implements hook_admin_menu_output_alter().
+ *
+ * @todo This is an addition, not an alter. Add a build hook stage.
+ */
+function admin_menu_admin_menu_output_alter(&$content) {
+  // Add "Add content" link tree on the top level.
+  $link = db_query("SELECT * FROM {menu_links} WHERE router_path = 'node/add' AND module = 'system'")->fetchAssoc();
+  $conditions = array();
+  for ($i = 1; $i < MENU_MAX_DEPTH; $i++) {
+    if (!empty($link["p$i"])) {
+      $conditions["p$i"] = $link["p$i"];
+    }
+  }
+  $tree = menu_build_tree($link['menu_name'], array(
+    'conditions' => $conditions,
+    'min_depth' => $link['depth'],
+  ));
+  $links = admin_menu_links_menu($tree);
+  if (!empty($links)) {
+    $key = key($links);
+    $links[$key]['#weight'] = -100;
+    $applied = FALSE;
+
+    //Try to add it inside the content menu.
+    foreach($content['menu'] as $k => $item) {
+      if($k[0] != '#') {
+        $parts = explode(" ", $k);
+        //Not sure how to reference the content menu
+        if(!$applied && !empty($parts[2]) && $parts[1] == 'Content' || $parts[2] == 9) {
+          $content['menu'][$k] += $links;
+          $applied = true;
+          break;
+        }
+      }
+    }
+    if (!$applied) {
+      $content['menu'] += $links;
+    }
+  }
+}
+
+/**
  * Render a themed list of links.
  *
  * @param $variables
